nav2_costmap_2d/include/nav2_costmap_2d/observation.hpp

```diff
@@ -89,6 +112,17 @@ class Observation
   {
   }
 
+  /**
+   * @brief  Move constructor
+   * @param obs The observation to move
+   */
+  Observation(Observation && obs)
```

Comment:
This seems like significant overkill. These are `double`s, using std::move on them is not necessary as far as I can tell

Reply:
Yes, looks like nothing here needs `std::move` for performance. Mostly here for uniformity and to show that everything is moving here.

> Added move assignment and move constructor, because it's they are called inside `std::vector` with less (de)allocations

That's why I think move constructor (as well as assignment) is needed, so there is no unnecessary PointCloud copies/allocations/deallocations every time some vector needs to grow.

Reply:
We want things to be created as needed and not have dead code in the system. If its not being used and isn't needed, then it should not be added. Its more code then we (I) need to maintain.